Classes/Telemetry/BITChannel.m

```diff
@@ -71,54 +70,74 @@ - (BOOL)isQueueBusy {
 
 - (void)persistDataItemQueue {
   [self invalidateTimer];
-  if (!BITSafeJsonEventsString || strlen(BITSafeJsonEventsString) == 0) {
+
+  char *jsonStream = NULL;
+
+  do {
+    jsonStream = BITSafeJsonEventsString;
+    @synchronized (self) {
```

Comment:
I don't now how to do it better, but we should change `BITSafeJsonEventsString` and `_dataItemCount` together